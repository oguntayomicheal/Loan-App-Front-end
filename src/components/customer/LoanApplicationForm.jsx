import React, { useState } from 'react';

import { NavLink, useNavigate } from 'react-router-dom';

import { FaSpinner } from 'react-icons/fa';
import '../../assets/styles/LoadingSpinner.css';

const LoanApplicationForm = () => {
  const navigate = useNavigate();
  const [loading, setLoading] = useState(false);
  const [loadingFailure, setLoadingFailure] = useState(false);

  const customerDetails = JSON.parse(localStorage.getItem('userData'));
  const customerID = customerDetails.id;

  const [formData, setFormData] = useState({
    loan_amount: '',
    purpose: '',
    repayment_preferences: '',
    customer_id: customerID,
    status: 'nil',
    details: 'nil',
    repayment_schedule: 'nil',
    amount_to_pay: 'nil'
  });

  const handleInputChange = (event) => {
    const { name, value } = event.target;
    setFormData({
      ...formData,
      [name]: value,
    });
  };

  const url = 'http://127.0.0.1:3100/api/v1/customers';

  const loanUrl = `${url}/loan_applications`;

  const handleSubmit = async (event) => {
    event.preventDefault();
    setLoading(true);
    setLoadingFailure(false);

    try {
      const response = await fetch(loanUrl, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ loanDetails: formData }),
      });

      if (response.ok) {
        // const data = await response.json();
        navigate('/LoanStatus');
      } else {
        setLoadingFailure(true);
        setLoading(false);
      }
    } catch (error) {
      throw new Error(error);
    }
  };

  return (

    <>
      <NavLink to="/LoanStatus" className="each_nav">
        Loan status
      </NavLink>

      <div>
        <h2>Loan Application Form</h2>
        <form onSubmit={handleSubmit}>
          <div className="form-group">
            <label htmlFor="loan_amount">
              Loan Amount ($$)
              <input
                type="number"
                id="loan_amount"
                name="loan_amount"
                placeholder="Enter desired loan amount"
                value={formData.loan_amount}
                onChange={handleInputChange}
                required
              />
            </label>

          </div>
          <div className="form-group">
            <label htmlFor="purpose">
              Loan Purpose
              <input
                type="textarea"
                id="purpose"
                name="purpose"
                placeholder="purpose"
                value={formData.purpose}
                onChange={handleInputChange}
                required
              />
            </label>
          </div>
          <div className="form-group">
            <label htmlFor="repayment_preferences">
              Repayment Preferences
              <input
                type="textarea"
                id="repayment_preferences"
                name="repayment_preferences"
                placeholder="Weekly, monthly..."
                value={formData.repayment_preferences}
                onChange={handleInputChange}
                required
              />
            </label>
          </div>
          <button type="submit">Submit Application</button>
        </form>

        {loading && (

        <div className="loading-spinner">
          <FaSpinner className="icon" />
        </div>

        )}

        {loadingFailure && (
        <div>
          <p className="login_error">Try again.</p>
        </div>
        )}
      </div>

    </>
  );
};

export default LoanApplicationForm;
