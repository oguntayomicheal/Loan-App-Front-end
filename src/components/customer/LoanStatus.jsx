import React, { useState, useEffect } from 'react';
import { NavLink, Link } from 'react-router-dom';

const LoanStatus = () => {
  const [loanApplicationsList, setLoanApplicationList] = useState([]);
  const customerData = localStorage.getItem('customerData');
  const customerId = (JSON.parse(customerData)).id;

  const loanUrl = `http://localhost:3100/api/v1/customers/${customerId}/loan_applications`;

  const fetchloanApplicationsList = async () => {
    fetch(loanUrl).then((response) => {
      if (!response.ok) {
        throw new Error('Network response not OK');
      }
      return response.json();
    }).then((data) => {
      setLoanApplicationList(data);
    });
  };

  useEffect(() => {
    fetchloanApplicationsList();
  }, []);

  return (
    <div className='loan_status_page'>
      <NavLink to="/LoanApplicationForm" className="each_nav">
        Application form
      </NavLink>
      <br />
      <div>
        <h2>Loan Status</h2>
        <ul id="allLoans">
          {loanApplicationsList.map((eachLoan) => (
            <li key={eachLoan.id} className="eachLoan">
              <span>
                <p>{eachLoan.loan_amount}</p>
              </span>
              <span>
                <p>Status</p>
                <p>
                  {' '}
                  {eachLoan.status}
                </p>
              </span>
              <span>
                <Link to={`/EachLoanStatus/${eachLoan.id}`}>
                  <button type="button">View</button>
                </Link>

              </span>
            </li>
          ))}
        </ul>
      </div>

    </div>
  );
};

export default LoanStatus;
