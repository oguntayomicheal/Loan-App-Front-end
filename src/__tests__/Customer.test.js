import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import LoanStatus from '../components/customer/LoanStatus';
import LoanApplicationForm from '../components/customer/LoanApplicationForm';

describe('Customer page', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  test('Renders loan status correctly', () => {
    localStorage.setItem('customerData', JSON.stringify({ id: 123 }));

    const loanStatus = render(
      <Router>
        <LoanStatus />
      </Router>,
    );
    expect(loanStatus).toMatchSnapshot();
  });

  test('Renders application form correctly', () => {
    localStorage.setItem('customerData', JSON.stringify({ id: 123 }));

    const loanApplicationForm = render(
      <Router>
        <LoanApplicationForm />
      </Router>,
    );
    expect(loanApplicationForm).toMatchSnapshot();
  });
});
