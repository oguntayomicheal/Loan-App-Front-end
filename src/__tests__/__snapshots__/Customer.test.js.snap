// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer loan status page Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loan_status_page"
      >
        <a
          class="each_nav"
          href="/LoanApplicationForm"
        >
          Application form
        </a>
        <br />
        <div>
          <h2>
            Loan Status
          </h2>
          <ul
            id="allLoans"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loan_status_page"
    >
      <a
        class="each_nav"
        href="/LoanApplicationForm"
      >
        Application form
      </a>
      <br />
      <div>
        <h2>
          Loan Status
        </h2>
        <ul
          id="allLoans"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Customer loan status page Renders correctly 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="customer_page main_content"
      >
        <a
          class="each_nav"
          href="/LoanStatus"
        >
          Loan status
        </a>
        <button
          class="logout_btn"
          type="button"
        >
          Logout
        </button>
        <div>
          <h2>
            Loan Application Form
          </h2>
          <form>
            <div
              class="form-group"
            >
              <label
                for="loan_amount"
              >
                Loan Amount ($$)
                <br />
                <input
                  id="loan_amount"
                  name="loan_amount"
                  placeholder="Enter desired loan amount"
                  required=""
                  type="number"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                for="purpose"
              >
                Loan Purpose
                <br />
                <input
                  id="purpose"
                  name="purpose"
                  placeholder="purpose"
                  required=""
                  type="textarea"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                for="repayment_preferences"
              >
                Repayment Preferences
                <br />
                <input
                  id="repayment_preferences"
                  name="repayment_preferences"
                  placeholder="Weekly, monthly..."
                  required=""
                  type="textarea"
                  value=""
                />
              </label>
            </div>
            <br />
            <button
              type="submit"
            >
              Submit Application
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="customer_page main_content"
    >
      <a
        class="each_nav"
        href="/LoanStatus"
      >
        Loan status
      </a>
      <button
        class="logout_btn"
        type="button"
      >
        Logout
      </button>
      <div>
        <h2>
          Loan Application Form
        </h2>
        <form>
          <div
            class="form-group"
          >
            <label
              for="loan_amount"
            >
              Loan Amount ($$)
              <br />
              <input
                id="loan_amount"
                name="loan_amount"
                placeholder="Enter desired loan amount"
                required=""
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              for="purpose"
            >
              Loan Purpose
              <br />
              <input
                id="purpose"
                name="purpose"
                placeholder="purpose"
                required=""
                type="textarea"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              for="repayment_preferences"
            >
              Repayment Preferences
              <br />
              <input
                id="repayment_preferences"
                name="repayment_preferences"
                placeholder="Weekly, monthly..."
                required=""
                type="textarea"
                value=""
              />
            </label>
          </div>
          <br />
          <button
            type="submit"
          >
            Submit Application
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Customer page Renders application form correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="customer_page main_content"
      >
        <a
          class="each_nav"
          href="/LoanStatus"
        >
          Loan status
        </a>
        <button
          class="logout_btn"
          type="button"
        >
          Logout
        </button>
        <div>
          <h2>
            Loan Application Form
          </h2>
          <form>
            <div
              class="form-group"
            >
              <label
                for="loan_amount"
              >
                Loan Amount ($$)
                <br />
                <input
                  id="loan_amount"
                  name="loan_amount"
                  placeholder="Enter desired loan amount"
                  required=""
                  type="number"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                for="purpose"
              >
                Loan Purpose
                <br />
                <input
                  id="purpose"
                  name="purpose"
                  placeholder="purpose"
                  required=""
                  type="textarea"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                for="repayment_preferences"
              >
                Repayment Preferences
                <br />
                <input
                  id="repayment_preferences"
                  name="repayment_preferences"
                  placeholder="Weekly, monthly..."
                  required=""
                  type="textarea"
                  value=""
                />
              </label>
            </div>
            <br />
            <button
              type="submit"
            >
              Submit Application
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="customer_page main_content"
    >
      <a
        class="each_nav"
        href="/LoanStatus"
      >
        Loan status
      </a>
      <button
        class="logout_btn"
        type="button"
      >
        Logout
      </button>
      <div>
        <h2>
          Loan Application Form
        </h2>
        <form>
          <div
            class="form-group"
          >
            <label
              for="loan_amount"
            >
              Loan Amount ($$)
              <br />
              <input
                id="loan_amount"
                name="loan_amount"
                placeholder="Enter desired loan amount"
                required=""
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              for="purpose"
            >
              Loan Purpose
              <br />
              <input
                id="purpose"
                name="purpose"
                placeholder="purpose"
                required=""
                type="textarea"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              for="repayment_preferences"
            >
              Repayment Preferences
              <br />
              <input
                id="repayment_preferences"
                name="repayment_preferences"
                placeholder="Weekly, monthly..."
                required=""
                type="textarea"
                value=""
              />
            </label>
          </div>
          <br />
          <button
            type="submit"
          >
            Submit Application
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Customer page Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loan_status_page"
      >
        <a
          class="each_nav"
          href="/LoanApplicationForm"
        >
          Application form
        </a>
        <br />
        <div>
          <h2>
            Loan Status
          </h2>
          <ul
            id="allLoans"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loan_status_page"
    >
      <a
        class="each_nav"
        href="/LoanApplicationForm"
      >
        Application form
      </a>
      <br />
      <div>
        <h2>
          Loan Status
        </h2>
        <ul
          id="allLoans"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Customer page Renders correctly 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="customer_page main_content"
      >
        <a
          class="each_nav"
          href="/LoanStatus"
        >
          Loan status
        </a>
        <button
          class="logout_btn"
          type="button"
        >
          Logout
        </button>
        <div>
          <h2>
            Loan Application Form
          </h2>
          <form>
            <div
              class="form-group"
            >
              <label
                for="loan_amount"
              >
                Loan Amount ($$)
                <br />
                <input
                  id="loan_amount"
                  name="loan_amount"
                  placeholder="Enter desired loan amount"
                  required=""
                  type="number"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                for="purpose"
              >
                Loan Purpose
                <br />
                <input
                  id="purpose"
                  name="purpose"
                  placeholder="purpose"
                  required=""
                  type="textarea"
                  value=""
                />
              </label>
            </div>
            <div
              class="form-group"
            >
              <label
                for="repayment_preferences"
              >
                Repayment Preferences
                <br />
                <input
                  id="repayment_preferences"
                  name="repayment_preferences"
                  placeholder="Weekly, monthly..."
                  required=""
                  type="textarea"
                  value=""
                />
              </label>
            </div>
            <br />
            <button
              type="submit"
            >
              Submit Application
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="customer_page main_content"
    >
      <a
        class="each_nav"
        href="/LoanStatus"
      >
        Loan status
      </a>
      <button
        class="logout_btn"
        type="button"
      >
        Logout
      </button>
      <div>
        <h2>
          Loan Application Form
        </h2>
        <form>
          <div
            class="form-group"
          >
            <label
              for="loan_amount"
            >
              Loan Amount ($$)
              <br />
              <input
                id="loan_amount"
                name="loan_amount"
                placeholder="Enter desired loan amount"
                required=""
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              for="purpose"
            >
              Loan Purpose
              <br />
              <input
                id="purpose"
                name="purpose"
                placeholder="purpose"
                required=""
                type="textarea"
                value=""
              />
            </label>
          </div>
          <div
            class="form-group"
          >
            <label
              for="repayment_preferences"
            >
              Repayment Preferences
              <br />
              <input
                id="repayment_preferences"
                name="repayment_preferences"
                placeholder="Weekly, monthly..."
                required=""
                type="textarea"
                value=""
              />
            </label>
          </div>
          <br />
          <button
            type="submit"
          >
            Submit Application
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Customer page Renders loan status correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loan_status_page"
      >
        <a
          class="each_nav"
          href="/LoanApplicationForm"
        >
          Application form
        </a>
        <br />
        <div>
          <h2>
            Loan Status
          </h2>
          <ul
            id="allLoans"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loan_status_page"
    >
      <a
        class="each_nav"
        href="/LoanApplicationForm"
      >
        Application form
      </a>
      <br />
      <div>
        <h2>
          Loan Status
        </h2>
        <ul
          id="allLoans"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
